File: app/src/main/java/com/maiya/call/util/context/ContextUtils.java
package com.maiya.call.util.context;

import android.app.Activity;
import android.content.Context;
import android.content.ContextWrapper;
import android.os.Build;

import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;

import com.maiya.call.util.LogUtils;

/**
 * Author : ymc
 * Date   : 2020/7/7  14:04
 * Class  : ContextUtils
 */
public class ContextUtils {

    public static boolean isDestroyed(Context context) {
        Activity activity = findActivity(context);
        if (activity == null) {
            return true;
        } else {
            boolean isDestroyed = false;
            if (Build.VERSION.SDK_INT >= 17) {
                isDestroyed = activity.isDestroyed();
            }

            if (activity instanceof FragmentActivity) {
                FragmentManager supportFragmentManager = ((FragmentActivity)activity).getSupportFragmentManager();
                isDestroyed = supportFragmentManager.isDestroyed();
            }

            LogUtils.i("activity.getClass().getName()>>" + activity.getClass().getName());
            return isDestroyed;
        }
    }

    public static Activity findActivity(Context context) {
        if (context == null) {
            return null;
        } else {
            if (context instanceof Activity) {
                return (Activity)context;
            } else if (context instanceof ContextWrapper) {
                ContextWrapper wrapper = (ContextWrapper)context;
                return findActivity(wrapper.getBaseContext());
            } else {
                return null;
            }
        }
    }

}

File: app/src/main/java/com/maiya/call/util/context/ContextUtils.kt
package com.maiya.call.util.context

import android.app.Activity
import android.content.Context
import android.content.ContextWrapper
import androidx.fragment.app.FragmentActivity
import com.maiya.call.util.LogUtils.i

/**
 * Author : ymc
 * Date   : 2020/7/7  14:04
 * Class  : ContextUtils
 */
object ContextUtils {
    fun isDestroyed(context: Context?): Boolean {
        val activity = findActivity(context)
        return if (activity == null) {
            true
        } else {
            var isDestroyed = activity.isDestroyed
            if (activity is FragmentActivity) {
                val supportFragmentManager = activity.supportFragmentManager
                isDestroyed = supportFragmentManager.isDestroyed
            }
            i("activity.getClass().getName()>>" + activity.javaClass.name)
            isDestroyed
        }
    }

    fun findActivity(context: Context?): Activity? {
        return if (context == null) {
            null
        } else {
            if (context is Activity) {
                context
            } else if (context is ContextWrapper) {
                findActivity(context.baseContext)
            } else {
                null
            }
        }
    }
}